numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4913,6 +4919,17 @@ initialize_static_globals(void)
         return -1;
     }
 
+    PyObject *_is_numpy2 = NULL;
+    npy_cache_import("numpy", "_numpy2_behavior", &_is_numpy2);
+    if (_is_numpy2 == NULL) {
+        return -1;
+    }
+    npy_numpy2_behavior = PyObject_IsTrue(_is_numpy2);
+    Py_DECREF(_is_numpy2);
+    if (npy_numpy2_behavior < 0) {
+        return -1;
```

Comment:
I think I would prefer there be one source of truth, and not have two global variables, one in C and one in Python

Reply:
OK, I guess that would make this `np._get_numpy2_behavior()` would add a bit of call overhead, but that probably doesn't really matter.  How does that sound?

Reply:
great, thanks

Reply:
Opted for `np._using_numpy2_behavior()`.  Threading it out from C is a bit weird (going through `numeric.py`), but maybe it doesn't matter.  I am now setting the variable directly in C.